### PR TITLE
fix: link active & hover not affecting Typography

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

Specifically targets the `Typography` class nested in the `Link` component

### Why is it needed?

Styles applied for `hover` and `active` were not affecting the text color accurately.

### Related issue(s)/PR(s)

CONTENT-477

### CODE REVIEW

File: packages/strapi-design-system/src/Link/Link.js [hunk 1]
```diff
@@ -12,18 +12,34 @@ const LinkWrapper = styled.a`
   align-items: center;
   text-decoration: none;
   pointer-events: ${({ disabled }) => (disabled ? 'none' : undefined)};
+
+  ${Typography} {
```
Comment (ronronscelestes) [inline packages/strapi-design-system/src/Link/Link.js:16]: 😍

File: packages/strapi-design-system/src/Link/Link.js [hunk 2]
```diff
@@ -12,18 +12,34 @@ const LinkWrapper = styled.a`
...
   }
+
   svg {
     font-size: ${10 / 16}rem;
   }
-  &:hover,
+
+  &:hover {
+    ${Typography} {
+      color: ${({ theme }) => theme.colors.primary500};
```
Comment (ronronscelestes) [inline packages/strapi-design-system/src/Link/Link.js:30]: should it not impact the icons too? 👀

image
image

Comment (maevalienard) [reply packages/strapi-design-system/src/Link/Link.js:30]: That would be nice indeed if feasible

Comment (joshuaellis) [reply packages/strapi-design-system/src/Link/Link.js:30]: I wasn't sure and thanks both for letting me know, i'll make those changes 😎

Comment (gu-stav) [reply packages/strapi-design-system/src/Link/Link.js]: You also may want to add `:focus,`?

File: packages/strapi-design-system/src/Link/Link.js [hunk 3]
```diff
@@ -12,18 +12,33 @@ const LinkWrapper = styled.a`
...
     font-size: ${10 / 16}rem;
   }
-  &:hover,
+
+  ${Typography} {
+    transition: color 150ms ease-out;
+    color: currentColor;
+  }
+
+  &:hover {
```
Comment (gu-stav) [inline packages/strapi-design-system/src/Link/Link.js:31]: Do you think we should add `&:focus` too?

**Review by ronronscelestes — APPROVED** [review]
LGTM (the transition ❤️)

**Pull request merged**